### PR TITLE
Language annotation support

### LINKED ISSUE #179: /* comments */ should not be reformatted
### Description

`/* */` style comments should not be reformatted. They have additional use cases such as tree-sitter picking up the the syntax to highlight before scripts inside multiline strings.

### Small example input

```nix
/* lua */ ''
  print("Hello, world!")
''
```

### Expected output

```nix
/* lua */ ''
  print("Hello, world!")
''
```

(now tree-sitter in my editor highlights this string to Lua which really helps me read/write code in this block now that it’s not just syntax highlighted as a string)

### Actual output

```nix
# lua
''
  print("Hello, world!")
''
```

### PR DESCRIPTION
**Pull request by dyegoaurelio**
Initial work to get language annotation support working

This pr is not ready, but I created it mostly to get feedback on how we should handle edge cases and the direction I'm taking.

closes #179

### CODE REVIEW

File: test/diff/language-annotation/in.nix [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  # Mixed comment styles
+  mixedComments = /* inline */ # line comment
+    ''
+      content
+    '';
+
+  # Language annotation in function arguments
+  processCode = builtins.readFile (/* lua */ ''
+    return "Hello"
+  '');
```
[the remark this answers is not in the document]
Comment (piegamesde) [reply test/diff/language-annotation/in.nix:64]: Yes, but also I can imagine inputs not covered by this PR which still exhibit this bug, so it might be worth tracking separately

**Comment by piegamesde** [comment]
Can you please outline the rough high-level approach taken here to make reading the code easier?

**Review by jfly — CHANGES_REQUESTED** [review]
LGTM, just one minor tweak, please!

File: src/Nixfmt/Lexer.hs [hunk 2]
```diff
@@ -156,17 +165,75 @@ isTrailing (PTBlockComment False []) = True
...
 isTrailing _ = False
 
-convertTrivia :: [ParseTrivium] -> Pos -> (Maybe TrailingComment, Trivia)
-convertTrivia pts nextCol =
+-- Check if a text is a valid language identifier for language annotations
+isLanguageIdentifier :: Text -> Bool
+isLanguageIdentifier content =
+  let stripped = strip content
+  in not (Text.null stripped)
+      && Text.length stripped <= 30 -- TODO: make configurable or remove limit
```
[the remark this answers is not in the document]
Comment (MattSturgeon) [reply src/Nixfmt/Lexer.hs]: Without a limit, we could end up with things like:

```nix
somewhat_long_name = /* very-long-comment-that-makes-this-line-so-long-you-cant-really-see-the-string-value */ "a moderate length string";
```

IIUC, the formatter will count that comment as having zero length when deciding whether to do any line wrapping.

Maybe there's a way we could have it so that normal comments are still not counted to the line length, but annotations are?

That sounds like the ideal solution, but I personally don't want to block this PR on it.
